template.tpl

```diff
@@ -243,21 +244,21 @@ const onFailure = () => {
   data.gtmOnFailure();
 };
 
-// Create the object to push to the data layer
-const pushData = cwvData => {
+// Create the object to push the Core Web Vitals Data to the data layer.
+const pushData = cwvObj => {
   
+  // Remove entries property, otherwise GTM will return null values, because the type of entries is not supported.
+  Object.delete(cwvObj, 'entries');
+  // Store the cleaned object in a new constant.
+  const cwvData = cwvObj;
+  
+  // Create the data layer object.
   const cwvDataLayerObject = {
     'event': eventName,
-    'webVitalsData': {
-      'delta': cwvData.delta,
-      'id': cwvData.id,
-      'name': cwvData.name,
-      'navigationType': cwvData.navigationType,
-      'rating': cwvData.rating,
-      'value': cwvData.value
-    }
+    'webVitalsData': cwvData
   };
   
+  // If user chooses attribution build, then add those data to a attribution property. Only supported types will be added.
```

Comment:
With the change above assigning `cwvData` directly to `cwvDataLayerObject.webVitalsData`, this will happen automatically all of the time.

Reply:
Thanks, good catch. Added a new commit.